### PR TITLE
feat: update flow to match new RollupCreator contract

### LINKED ISSUE #32: Adjust deployment flow for the new BridgeCreator contract
We can now take advantage of the new `BridgeCreator` contract at  [0x40Ae6427825f6be28F753aBD72f9A3f36b09eCb9](https://goerli.arbiscan.io/address/0x40Ae6427825f6be28F753aBD72f9A3f36b09eCb9), which allows us to do everything (deploy rollup, set validators and batch poster) within a single transaction.

This requires some changes in the UI, but in general we can keep the flow pretty much the same, just submit the transaction at the 4th step (`Review & Deploy`). This would be the new list of all steps:

```
Configure Rollup -> Configure Validators -> Configure Batch Poster -> Review & Deploy -> Review & Download Config
```

It also probably makes sense to have `Next` and `Previous` buttons to go between the first three steps.

### PR DESCRIPTION
**Pull request by douglance**
closes #32 

- updates rollup to new contract that configures rollup, validators, and batch poster in one tx
- updates deployment flow 
- makes batch poster editable and defaults to empty string
- makes validator default to empty string
- adds `Next` and `Previous` buttons to navigate through UI
- moves `Reset` button to above forms

### CODE REVIEW

File: src/utils/deployRollup.ts [hunk 1]
```diff
@@ -205,9 +226,14 @@ export async function deployRollup({
     'deployed-at': rollupContracts.deployedAtBlockNumber,
   };
 
-  l3Config.rollup= rollupCreatedEvent.args.rollupAddress;
+  rollupConfigData.node['batch-poster']['parent-chain-wallet']['private-key'] =
```
Comment (spsjvc) [inline src/utils/deployRollup.ts:229]: Let's make sure the `l3Config` is also updated: https://github.com/OffchainLabs/orbit-deployment-ui/blob/main/src/components/SetBatchPoster.tsx#L42

File: src/utils/deployRollup.ts [hunk 2]
```diff
@@ -205,9 +226,14 @@ export async function deployRollup({
...
+    batchPoster.privateKey || '';
+  rollupConfigData.node.staker['parent-chain-wallet']['private-key'] =
```
Comment (spsjvc) [inline src/utils/deployRollup.ts:231]: Let's make sure the `l3Config` is also updated: https://github.com/OffchainLabs/orbit-deployment-ui/blob/main/src/components/SetValidators.tsx#L45C1-L45C1

**Comment by GreatSoshiant** [comment]
If you click on "Restart and Start Over", the owner field wouldn’t default to the connected wallet address, instead showing up empty field

File: src/components/ResetButton.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  function reset() {
+    localStorage.removeItem('rollupData');
+    localStorage.removeItem('l3Config');
+
+    dispatch({ type: 'reset' });
+
+    router.push('/deployment?step=1');
+  }
+
```
Comment (GreatSoshiant) [inline src/components/ResetButton.tsx:16]: If you click on restart and start over, the owner field wouldn’t default to the connected wallet address, instead it'll be empty. not sure how to resolve it

**Review by spsjvc — APPROVED** [review]
Nice, feel free to merge after the 2 comments are addressed

**Pull request merged**